src/zarr/codecs/bytes.py

```diff
@@ -71,14 +76,12 @@ async def _decode_single(
         chunk_spec: ArraySpec,
     ) -> NDBuffer:
         assert isinstance(chunk_bytes, Buffer)
-        if chunk_spec.dtype.itemsize > 0:
-            if self.endian == Endian.little:
-                prefix = "<"
-            else:
-                prefix = ">"
-            dtype = np.dtype(f"{prefix}{chunk_spec.dtype.str[1:]}")
-        else:
-            dtype = np.dtype(f"|{chunk_spec.dtype.str[1:]}")
+        # TODO: remove endianness enum in favor of literal union
+        endian_str = cast(
+            "Endianness | None", self.endian.value if self.endian is not None else None
```

Comment:
Maybe I'm going mad, but can't this just be `self.endian.value` and the `None` checking can be dropped?

Reply:
`self.endian` can be `None`, so we have to check for that before we can retrieve the value. IMO there's no reason to use the `endian` enum and we could just use a literal type but I didn't want to make that change in this PR.

Reply:
Ah I missed the difference between `self.endian.value` and `self.endian`, sorry!